### PR TITLE
Allow multiple delimiters on Inflector::humanize

### PR DESCRIPTION
**Pull request by ahmyi**
Allow mutiple delimiters for Inflector::humanize method
eg.
```PHP
$field = "user.is_active";
echo Inflector::humanize($field,[".","_"]); 
```
Will return User Is Active instead of User.is_active or User Is_active

### CODE REVIEW

File: src/Utility/Inflector.php [hunk 1]
```diff
@@ -625,7 +625,7 @@ public static function underscore($string)
      * Also replaces underscores with dashes
      *
      * @param string $string The string to dasherize.
-     * @return string Dashed version of the input string
+     * @return mixed Dashed version of the input string
```
Comment (dereuromark) [inline src/Utility/Inflector.php]: This looks like an invalid change.

File: src/Utility/Inflector.php [hunk 2]
```diff
@@ -625,7 +625,7 @@ public static function underscore($string)
      * Also replaces underscores with dashes
      *
      * @param string $string The string to dasherize.
-     * @return string Dashed version of the input string
+     * @return string|array Dashed version of the input string or multiple strings
```
Comment (dereuromark) [inline src/Utility/Inflector.php]: Why do you think this? It is not true. This is still an array.

File: src/Utility/Inflector.php [hunk 3]
```diff
@@ -643,8 +643,12 @@ public static function dasherize($string)
      */
     public static function humanize($string, $delimiter = '_')
```
Comment (dereuromark) [inline src/Utility/Inflector.php]: If you accept array, why not changing the doc block?

File: src/Utility/Inflector.php [hunk 4]
```diff
@@ -643,8 +643,12 @@ public static function dasherize($string)
...
     {
-        $cacheKey = __FUNCTION__ . $delimiter;
-
+        $delimiters = $delimiter;
+        if(is_array($delimiter)) {
```
Comment (dereuromark) [inline src/Utility/Inflector.php]: CS invalid here

**Comment by dereuromark** [comment]
I am not sure if weakening the input type to allow for array use case is a good idea for the existing method.
Especially not until 4.x.

Maybe this just warrants its own (plugin/app or core) method instead?

File: src/Utility/Inflector.php (hunk 2 above)
Comment (ahmyi) [reply src/Utility/Inflector.php]: Yes. In cases like having multiple delimiters an array should be passed. As test case given  

```PHP
$this->assertSame('User Is Active', Inflector::humanize('user.is_active',['.','_']));
```
If you pass an array directly you will have a notice error at 
```PHP
 $cacheKey = __FUNCTION__ . $delimiter;
```

File: src/Utility/Inflector.php (hunk 3 above)
Comment (ahmyi) [reply src/Utility/Inflector.php]: Sorry wrong doc block

**Comment by ahmyi** [comment]
Use case for Forms for with multiple models. When label with 
```PHP
$this->Form->controller(<model>.<field>); 
```
It will display Name as label for multiple fields with name. Example
```PHP
echo $this->Form->control('Father.first_name',['type'=>'text']);
echo $this->Form->control('Mother.first_name',['type'=>'text']);
```
image

Implementing in FormHelper with humanize multiple delimiters the output will be different

image


Well user can use if you want to strict with only string

```PHP
echo Inflector::humanize(Inflector::humanize($field,"."))
```

**Comment by dereuromark** [comment]
Instead of master, should this point to 3.next (3.7?)

**Comment by markstory** [comment]
You have mentioned that this will make form label inflection better but I don't see where those changes are.

On it's own, I don't see how this is better than calling humanize() twice. If an application needs this kind of inflection in multiple places a function could be defined to provide the desired results.

**Comment by dereuromark** [comment]
Do you find a way to have this part in the FormHelper instead?
The general direction of feedback seems to lean in that direction instead of changing this low level method here.

**Comment by dereuromark** [comment]
Closing in favor of discussion https://github.com/cakephp/cakephp/issues/12702 for now, lets see what new solution we can come up with.

**Pull request closed without merging**